extension/observer/ecsobserver/fetcher.go

```diff
@@ -219,6 +222,19 @@ func (f *taskFetcher) attachTaskDefinition(ctx context.Context, tasks []*ecs.Tas
 	return tasksWithDef, nil
 }
 
+// filterDiscoverableTasks preserves only fargate tasks or EC2 task which container
+// instance is known.
+func (f *taskFetcher) filterDiscoverableTasks(tasks []*taskAnnotated) []*taskAnnotated {
+	var result []*taskAnnotated
+	for _, t := range tasks {
+		// Preserve only fargate tasks or EC2 tasks with non-nil ContainerInstanceArn
+		if t.Task.ContainerInstanceArn != nil || aws.StringValue(t.Task.LaunchType) != ecs.LaunchTypeEc2 {
```

Comment:
I think this condition can be moved to `getAllTasks` given that that function isn't used anywhere else. `filterDiscoverableTasks` won't be needed then, but `getAllTasks` should be renamed to something like `discoverableTasks`. It'll save cpu/memory on unnecessary processing and allocations and reduce the code without affecting readability.